tests/dataframe/unit/test_functions.py

```diff
@@ -1188,8 +1188,14 @@ def test_bin(self):
 
     def test_hex(self):
         col_str = SF.hex("cola")
-        self.assertEqual("HEX(cola)", col_str.sql())
+        self.assertEqual("LOWER(HEX(cola))", col_str.sql())
```

Comment:
why did this test change?

Reply:
Because `col_str`  is a exp.Hex (lower hex), but when we call sql without dialect, it will use `SQLGlot` dialect as the default. And default of  HEX_LOWERCASE is False, so exp.Hex will produce `LOWER(HEX(x))` instead. 

So, I made another test for upperhex which work exactly like this old test:
https://github.com/tobymao/sqlglot/blob/ac4bc7e4447c818b42da7c8250a7713fcf1afb12/tests/dataframe/unit/test_functions.py#L1195-L1200

Reply:
> it will use SQLGlot dialect as the default.

`col_str` is not a plain sqlglot `Expression` though, it's a dataframe `Column` object which is [generated into Spark by default](https://github.com/tobymao/sqlglot/blob/58d5f2bece42acdda5f8c08d30e6f61a5e538d4c/sqlglot/dataframe/sql/column.py#L214). So something seems off, I don't think this test should've changed?

Reply:
Hmm thank you for correcting me, it is Spark dialect, not SQLGlot dialect.

But, spark, like SQLGlot, is not a HEX_LOWERCASE. `SF.hex("cola")` create a Column which contains a Hex expression directly (stored in self.expression), and exp.Hex is treated as LowerHex, so the test will fail if unchanged. 

```python
from sqlglot import parse_one, exp
from sqlglot.dialects.spark import Spark

spark_dialect = Spark()

# normal transpiler
parse_one("HEX(cola)").sql(dialect=spark_dialect)
# 'HEX(cola)'

# Create exp.Hex directly
exp.Hex(this="cola").sql(dialect=spark_dialect)
# 'LOWER(HEX(cola))'
```

Can you suggest me where we should change to pass this test, if without the modified above?

Reply:
> But, spark, like SQLGlot, is not a HEX_LOWERCASE [...] and exp.Hex is treated as LowerHex, so the test will fail if unchanged.

The issue is that you're treating it as a `LowerHex` even though the default setting is treating `HEX` as "uppercase". See my other comment.